scripts/pancancer.py

```diff
@@ -0,0 +1,341 @@
+                            {
+                                'name':'dkfz_dockered_workflows',
+                                'url':'https://s3.amazonaws.com/oicr.docker.private.images/dkfz_dockered_workflows_1.3.tar'
+                            }
+                        },
+                        'ami_id':'ami-12345',
+                        'default-ini':'http://something.ini',
+                        'instance-type':'m3.medium',
+                        'lvm_devices':'/dev/xvdb,/dev/xvdc,/dev/xvdd,/dev/xvde'
+                    }
+                }
+
+    @staticmethod
+    def get_workflow_names():
+        keys = ''
+        for k in WorkflowLister._workflows:
+            keys += k+'\n'
+        return keys
+
+    @staticmethod
+    def get_workflow_details(workflow_name):
+        return WorkflowLister._workflows[workflow_name]
+
+###
+
+class Workflows(Command):
+    "This command  can help you configure and select workflows."
+    log = logging.getLogger(__name__)
+
+    def get_parser(self,prog_name):
+        parser = super(Workflows,self).get_parser(prog_name)
+        #parser.add_mutually_exclusive_group()
+        workflows_subparser = parser.add_subparsers(title='subcommands',help='workflows subcommands: list and config',dest='subparser_name')
+        workflows_subparser.add_parser('list',help='Get a list of workflows')
+        workflows_subparser.add_parser('config',help='Generate a default config file for a specific workflow').add_argument('workflow_name',help='Name of workflow to configure')
+        return parser
+
+    def take_action(self, parsed_args):
+        subparser_name=vars(parsed_args)['subparser_name']
+        self.log.debug('subparser: %s',subparser_name)
+        if subparser_name=='list':
+            workflow_list = WorkflowLister.get_workflow_names()
+            print ('Available workflows are:')
+            print(workflow_list)
+
+        elif subparser_name=='config':
+            workflow_name=vars(parsed_args)['workflow_name']
+            if workflow_name in WorkflowLister.get_workflow_names():
+                # TODO: Actually generate the correct INI file!
+                print('generating default INI in /home/ubuntu/ini-dir/'+workflow_name+'.ini for workflow '+workflow_name)
+            else:
+                print ('Sorry, but '+workflow_name+' is not a valid workflow name. Please use the command \'workflows list\' to see a list of available workflows.')
+        else:
+            print ('The available subcommands are \'list\' and \'config\'.')
+
+###
+
+class DaemonCommand(Command):
+    "Parent class for commands that start/stop daemons"
+    service_name=''
+    log = logging.getLogger(__name__)
+
+    def _do_start(self,start_cmd):
+        self.log.debug (start_cmd)
+        p = subprocess.Popen(start_cmd.split(' '))
+        # Wait a moment... if we get ANY response, it means that the process did not start, probably because it's already been started.
+        time.sleep(2)
```

Comment:
The sleep was added because I found that if I did `p.poll()` immediately, I would not get the result of the `flock...` command. I'm trying to find a better way to control these processes. It would have been much easier if ansible hadn't had problems when the Deployer was called from the service script.